platform/podjs/bin/genPodjs.js

```diff
@@ -0,0 +1,75 @@
+const MagicString = require("magic-string");
+const fs = require("fs");
+
+function replaceManifestData(code, manifestData) {
+    const magicString = new MagicString(code);
+    const pattern = new RegExp("window.manifestData", "g");
+
+    let hasReplacements = false;
+    let match;
+    let start;
+    let end;
+    let replacement;
+
+    while ((match = pattern.exec(code))) {
+        hasReplacements = true;
+
+        start = match.index;
+        end = start + match[0].length;
+
+        replacement = String(manifestData);
+
+        magicString.overwrite(start, end, replacement);
+    }
+
+    if (!hasReplacements) {
+        return null;
+    }
+
+    return magicString.toString();
```

Comment:
Not clear why you're using `magic-string` here. If all you want to do is replace, you can simply do so via `replace`. `magic-string` is apparently useful if you want to generate source maps, but does not seem to be the case here, as far as I can tell. You don't even need a regex here, `replace` will replace one string into another. Is there something I'm missing here?

Reply:
well, it was supposed to be a rollup plugin thus I needed magic-string. Can try without it.

Reply:
Is there any specific functionality magic-string provides in this case regarding rollup? As I say above, apparently it can generate source maps, but I'm not totally sure we're using that functionality in this case?